src/Forms/Main.cs

```diff
@@ -26677,8 +26678,46 @@ private void toolStripMenuItemDost_Click(object sender, EventArgs e)
 
         private void toolStripMenuItemMeasurementConverter_Click(object sender, EventArgs e)
         {
-            var form = new MeasurementConverter();
-            form.Show(this);
+            var selectedText = string.Empty;
+            if (tabControlSubtitle.SelectedIndex == TabControlSourceView)
+            {
+                selectedText = textBoxSource.SelectedText;
+            }
+            else
+            {
+                if (textBoxListViewText.Focused)
+                {
+                    selectedText = textBoxListViewTextAlternate.SelectedText;
+                }
+                else
+                {
+                    selectedText = textBoxListViewText.SelectedText;
+                }
+            }
+
+            if (_measurementConverter != null && !_measurementConverter.IsDisposed)
+            {
+                _measurementConverter.WindowState = FormWindowState.Normal;
+                _measurementConverter.Input = selectedText;
+                _measurementConverter.Focus();
+                return;
+            }
+
+            _measurementConverter = new MeasurementConverter
+            {
+                Input = selectedText
+            };
+            _measurementConverter.Show(this);
+
+            _measurementConverter.ReplaceClicked += measurementConverter_ReplaceClicked;
+        }
+
+        private void measurementConverter_ReplaceClicked(object sender, MeasurementConverter.ReplaceEventArgs e)
+        {
+            if (IsSubtitleLoaded)
+            {
+                textBoxListViewText.SelectedText = e.Result; 

```

Comment:
Should also know which textbox to use.

Reply:
I was thinking it should always use the main one, you don't usually change the units in the original file. 

What do you think? 
Also, if you think it should be the textbox that was focused, how would I get that? Since it loses focus when switching to the converter.

Reply:
I think it should work for both boxes. You can save flag in the "measurement converter" regarding which box was active.
Also check the  `GetFocusedTextBox()` method.

Reply:
Could you also merge `master` into your branch? (there's a conflict in Settings.cs)

Reply:
I solved the conflict. :D
It was the "Remove unneeded initialization".